snoop/data/tasks.py

```diff
@@ -183,6 +184,15 @@ def run_task(task_pk, raise_exceptions=False):
                 log=handler.stream.getvalue(),
             )
 
+        except ConnectionError as e:
+            logger.exception(repr(e))
```

Comment:
Again, `logger.exception` prints all the exception details, you should supply a human-readable message explaining when the error occurred.